### PR TITLE
hatch-1.3.1/hatchling-1.6.0

### PR DESCRIPTION
**Pull request by jcgruenhage**
#### Testing the changes
- I tested the changes in this PR: **briefly**

I've tried building and testing all packages that depend on hatchling, and except for hatch everything went smoothly. The test failures in hatch don't look very problematic to me though. Should we just deselect those?

Log: https://pastebin.com/raw/uywdGiZY

### CODE REVIEW

**Comment by icp1994** [comment]
Shouldn't [python3-editables](https://pypi.org/project/editables/) be a [dependency](https://github.com/pypa/hatch/blob/hatchling-v1.4.1/backend/src/hatchling/ouroboros.py#L36) for `hatchling`? That one is not packaged yet it seems.

**Comment by jcgruenhage** [comment]
yeah, should be, although it seems unrelated to the problem

**Comment by jcgruenhage** [comment]
(As in: I tested it locally again with the update to the newer hatch/hatchling versions and there the tests passed again, and only after that I also added the editables dep)

**Comment by icp1994** [comment]
Ah I see the editable tests were already disabled via `make_check_args`

**Comment by jcgruenhage** [comment]
yeah... ._.

I've tried re-enabling them after I had added the editables dep, but they still seem to fail. I'll try to figure out why they still fail

File: srcpkgs/python3-editables/template [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Template file for 'python3-editables'
+pkgname=python3-editables
+version=0.3
+revision=1
+wrksrc="${pkgname/python3-/}-${version}"
+build_style=python3-pep517
+hostmakedepends="python3-setuptools python3-wheel"
+depends="python3"
+checkdepends="python3-pip python3-coverage python3-pytest-cov python3-pytest python3-virtualenv"
```
Comment (classabbyamp) [inline srcpkgs/python3-editables/template:9]: why does this need pip and virtualenv?

Comment (jcgruenhage) [reply srcpkgs/python3-editables/template]: I'm pretty sure there was a reason I added this, but I don't know anymore, and the tests of `editables` seem to pass without it, so I've removed them again.

**Comment by jcgruenhage** [comment]
So, I've updated this again, and there's been new tests failing that weren't among the failed tests yet. I haven't had the time to investigate this, but I don't want to have a longer and longer list of failures here, so in case anyone else has the time to help investigate the test failures building up for `hatch`, I'd appreciate it.

**Comment by icp1994** [comment]
After some tinkering around, I think I have figured out what's causing the new errors. I'll list the reasoning sequentially for easier reading.

1. The tests for `hatchling` comes bundled with the `hatch` source in `tests/backend` folder.
2. At the time of `hatch-1.3.1` release, the latest release of `hatchling` was `1.5.0`. The `tests/backend` inside `hatch-1.3.1.tar.gz` were therefore meant to run alongside `hatchling-1.5.0`.
3. But with the current template, building `hatch-1.3.1` pulls `hatchling-1.6.0` as dependency so some of the old tests which aren't compatible anymore with the newer `hatchling` version fails.

We should be able to merge `hatchling-1.5.0` and `hatch-1.3.1` first and then `hatchling-1.6.0` in a separate commit since the `hatchling` package runs no tests. Or we could decouple `hatchling` tests from `hatch` so this headache doesn't occur in the future whenever we skip a couple minor updates and the tests become incompatible.

**Comment by jcgruenhage** [comment]
Thanks for doing the detective work, totally makes sense! I'm currently trying what happens if I just ignore `backend/tests` and `tests/backend` in `hatch`, then it should work just fine. I prefer this over carefully updating the versions in a specific order, because that'll cause headaches in the future as well.

I'd still like to get the tests in `hatchling` to work though, so that these tests are only ignored when building `hatch`, but actually are being run when building `hatchling`.

**Comment by jcgruenhage** [comment]
Let's see what the CI says, but it builds successfully locally, tests pass correctly!

**Pull request merged**